tests/HUBViewControllerTests.m

```diff
@@ -3101,6 +3101,58 @@ - (void)testThatModelIsNotSetImmediatelyOnOverlappingRenderRequests
     viewModelRenderer.completionBlocks[1]();
 }
 
+- (void)testThatModelIsNotSetImmediatelyOnOverlappingRenderAndLayoutRequests
+{
+    HUBViewModelRendererMock *viewModelRenderer = [HUBViewModelRendererMock new];
+    [self createViewControllerWithViewModelRenderer:viewModelRenderer];
+
+    self.contentOperation.contentLoadingBlock = ^(id<HUBViewModelBuilder> viewModelBuilder) {
+        [viewModelBuilder headerComponentModelBuilder].title = @"title1";
+        return YES;
+    };
+
+    [self simulateViewControllerLayoutCycle];
+
+    // The model should be set at this point as there's no existing render in progress
+    XCTAssertEqualObjects(self.viewController.viewModel.headerComponentModel.title, @"title1");
+
+    self.contentOperation.contentLoadingBlock = ^(id<HUBViewModelBuilder> viewModelBuilder) {
+        [viewModelBuilder headerComponentModelBuilder].title = @"title2";
+        return YES;
+    };
+
+    // While the 1st render is in progress, a 2nd render request comes in
+    [self.viewController reload];
+
+    // The 2nd model should NOT be set at this point as the 1st render hasn't finished
+    XCTAssertEqualObjects(self.viewController.viewModel.headerComponentModel.title, @"title1");
+    // Only 1 render request has been made at this point (the 2nd is pending)
+    XCTAssertEqual(viewModelRenderer.completionBlocks.count, (NSUInteger)1);
+    // Complete the 1st render request
+    viewModelRenderer.completionBlocks[0]();
+
+    // The 2nd model should now be set as the 1st render has finished and the 2nd is in progress
+    XCTAssertEqualObjects(self.viewController.viewModel.headerComponentModel.title, @"title2");
+    // 2 render requests have now been made
+    XCTAssertEqual(viewModelRenderer.completionBlocks.count, (NSUInteger)2);
+
+    // While the 2nd render is in progress, a layout requests comes in.
+    CGRect rect = self.viewController.view.bounds;
+    CGRect offsetRect = CGRectOffset(rect, 1, 1);
+    self.viewController.view.bounds = offsetRect;
+    [self.viewController.view layoutIfNeeded];
```

Comment:
Wouldn't it be better to call `[viewController layoutSubviews]` directly? That way you're sure of the execution flow, so the test becomes more predictable.